lib/base_service.ts

```diff
@@ -192,18 +191,42 @@ export class BaseService {
     }
   }
 
+  /**
+   * Guarantee that the next request you make will be IAM authenticated. This
+   * performs any requests necessary to get a valid IAM token so that if your
+   * next request involves a streaming operation, it will not be interrupted.
+   *
+   * @param {Function} callback - callback function to return flow of execution
+   *
+   * @returns {void}
+   */
+  protected preAuthenticate(callback): void {
+     if (Boolean(this.tokenManager)) {
+      return this.tokenManager.getToken((err, token) => {
+        if (err) {
+          callback();
+        }
+        const authHeader = { Authorization: 'Bearer ' + token };
+        this._options.headers = extend(authHeader, this._options.headers);
+        callback();
+      });
+    } else {
+      callback();
+    }
+  }
+
```

Comment:
why do you do this in a method? I don't like having a method that changes a private variable like `_options()`

Reply:
I actually didn't mean to leave that in there, that was mostly  to account for `createRecognizeStream`, which now has a different solution. There is still an issue of the way `recognize` handles headers. It puts them in `parameters.options` rather than `parameters.defaultOptions` like all the generated methods.

This leads to the code crashing when it gets here:
```js
        parameters.defaultOptions.headers.Authorization =
          `Bearer ${accessToken}`;
```

Can we change `recognize` to put the headers in `defaultOptions` to stay consistent? Otherwise, I will add logic like this to `createRequest` in the base service:
```js
if (parameters.defaultOptions.headers) {
  parameters.defaultOptions.headers.Authorization = `Bearer ${accessToken}`;
} else {
  parameters.options.headers.Authorization = `Bearer ${accessToken}`;
}
```

Let me know what you think @germanattanasio

Reply:
I don't have a preference here so I will let you decide